brainsets_pipelines/kemp_sleep_edf_2013/pipeline.py

```diff
@@ -0,0 +1,404 @@
+parser.add_argument("--reprocess", action="store_true")
+parser.add_argument(
+    "--study_type",
+    type=str,
+    choices=["sc", "st", "both"],
+    default="both",
+    help="Which study to download: 'sc' (Sleep Cassette), 'st' (Sleep Telemetry), or 'both'",
+)
+
+
+class Pipeline(BrainsetPipeline):
+    brainset_id = "kemp_sleep_edf_2013"
+    parser = parser
+
+    @classmethod
+    def get_manifest(cls, raw_dir: Path, args) -> pd.DataFrame:
+        s3 = get_s3_client()
+
+        prefixes = []
+        if args.study_type in ["sc", "both"]:
+            prefixes.append(f"{PREFIX}/sleep-cassette/")
+        if args.study_type in ["st", "both"]:
+            prefixes.append(f"{PREFIX}/sleep-telemetry/")
+
+        manifest_rows = []
+
+        for prefix in prefixes:
+            paginator = s3.get_paginator("list_objects_v2")
+            for page in paginator.paginate(Bucket=BUCKET, Prefix=prefix):
+                for obj in page.get("Contents", []):
+                    key = obj["Key"]
+                    if key.endswith("PSG.edf"):
+                        filename = Path(key).name
+                        base_name = filename.replace(".edf", "")
+
+                        hypnogram_key = cls._find_hypnogram_key(s3, key)
+
+                        if "sleep-cassette" in key:
+                            subject_id = base_name[3:5]
+                            study_type = "sleep_cassette"
+                        elif "sleep-telemetry" in key:
+                            subject_id = base_name[3:5]
+                            study_type = "sleep_telemetry"
+                        else:
+                            subject_id = base_name
+                            study_type = "unknown"
+
+                        session_id = f"{study_type}_{base_name}"
+
+                        manifest_rows.append(
+                            {
+                                "session_id": session_id,
+                                "psg_s3_key": key,
+                                "hypnogram_s3_key": hypnogram_key,
+                                "study_type": study_type,
+                                "subject_id": subject_id,
+                            }
+                        )
+
+        manifest = pd.DataFrame(manifest_rows).set_index("session_id")
+        return manifest
+
+    @classmethod
+    def _find_hypnogram_key(cls, s3, psg_key: str) -> Optional[str]:
+        """Find the hypnogram file corresponding to a PSG file."""
+        psg_path = Path(psg_key)
+        base_name = psg_path.stem
+
+        for prefix_len in [7, 6]:
+            prefix = base_name[:prefix_len]
+            search_prefix = str(psg_path.parent / prefix)
+
+            response = s3.list_objects_v2(Bucket=BUCKET, Prefix=search_prefix)
+
+            for obj in response.get("Contents", []):
+                if "Hypnogram" in obj["Key"] and obj["Key"].endswith(".edf"):
+                    return obj["Key"]
+
+        return None
```

Comment:
Maybe nit: Instead of having this as a `classmethod`, you could define this as a function instead the `get_manifest` classmethod?